plugins/GeoIp2/GeoIP2AutoUpdater.php

```diff
@@ -186,8 +200,12 @@ public static function unzipDownloadedFile($path, $dbType, $unlink = false)
             $fileToExtract = null;
             foreach ($content as $info) {
                 $archivedPath = $info['filename'];
-                if (in_array(basename($archivedPath), LocationProviderGeoIp2::$dbNames[$dbType])) {
-                    $fileToExtract = $archivedPath;
+                foreach (LocationProviderGeoIp2::$dbNames[$dbType] as $dbName) {
+                    if (basename($archivedPath) === $dbName
+                        || preg_match('/' . $dbName . '/', basename($archivedPath))
```

Comment:
You might want to use `preg_quote($dbName, '/')  ` just to be safe?

Reply:
It's not supposed to have preg_quote, the values in `$dbNames` are either the file name or a regex, so they should already be quoted appropriately, or they're just string names. I could separate them into two static variables perhaps...

Reply:
It's just that you need to know how the regex is escaped where the regexes are defined. Like where you define the regexes is different place to where the preg_match is defined and by using preg_quote you don't need to know it uses a `slash`. It's a detail though not too important.

Reply:
Yes, but that would also make a regex like `dbip-city-lite-\d{4}-\d{2}.mmdb` match the exact string `"dbip-city-lite-\d{4}-\d{2}.mmdb"` instead of match as a regex. Or am I missing something?

Reply:
True... it be more like a `str_replace('/', \/, $regex)`... not needed though